src/information_panel.rs

```diff
@@ -0,0 +1,459 @@
+        } else {
+            None
+        }
+    }
+
+    /// Renders the Information Panel in the provided UI context.
+    ///
+    /// # Arguments
+    /// - `ui`: The UI context where the panel will be rendered.
+    /// - `file_dialog`: A reference to the current file dialog, which provides the active file entry.
+    pub fn ui(&mut self, ui: &mut Ui, file_dialog: &mut FileDialog) {
+        const SPACING_MULTIPLIER: f32 = 4.0;
+
+        ui.label("Information");
+        ui.separator();
+
+        // Display metadata in a grid format
+        let width = file_dialog.config_mut().right_panel_width.unwrap_or(100.0) / 2.0;
+
+        if let Some(item) = file_dialog.active_entry() {
+            // load file content and additional metadata if it's a new file
+            self.load_meta_data(item);
+
+            // show preview of selected item
+            self.display_preview(ui, item);
+
+            let spacing = ui.ctx().style().spacing.item_spacing.y * SPACING_MULTIPLIER;
+            ui.separator();
+
+            ui.add_space(spacing);
+
+            // show all metadata
+            self.display_meta_data(ui, width, item);
+        }
+    }
+
+    fn display_preview(&mut self, ui: &mut Ui, item: &DirectoryEntry) {
+        if item.is_dir() {
+            // show folder icon
+            ui.vertical_centered(|ui| {
+                ui.label(egui::RichText::from(item.icon()).size(120.0));
+            });
+        } else {
+            // Display file content preview based on its extension
+            if let Some(ext) = item.as_path().extension().and_then(|ext| ext.to_str()) {
+                if let Some(panel_entry) = &self.panel_entry {
+                    if let Some(preview_handler) =
+                        self.supported_preview_files.get_mut(&ext.to_lowercase())
+                    {
+                        preview_handler(ui, panel_entry);
+                    } else if let Some(preview_handler) =
+                        self.supported_preview_images.get_mut(&ext.to_lowercase())
+                    {
+                        preview_handler(ui, panel_entry, &mut self.stored_images);
+                        let number_of_stored_images = self.stored_images.len();
+                        if number_of_stored_images > 10 {
+                            self.forget_last_stored_image(ui);
+                        }
+                    } else if let Some(mut content) = panel_entry.content() {
+                        egui::ScrollArea::vertical()
+                            .max_height(100.0)
+                            .show(ui, |ui| {
+                                ui.add(egui::TextEdit::multiline(&mut content).code_editor());
+                            });
+                    } else {
+                        // if now preview is available, show icon
+                        ui.vertical_centered(|ui| {
+                            ui.label(egui::RichText::from(item.icon()).size(120.0));
```

Comment:
The size should not be hard coded but should be determined by the size available